storm-core/test/jvm/org/apache/storm/utils/UtilsTest.java

```diff
@@ -0,0 +1,213 @@
+            conf.put(Config.STORM_EXHIBITOR_SERVERS,"foo");
+            conf.put(Config.STORM_EXHIBITOR_PORT, 0);
+            conf.put(Config.STORM_EXHIBITOR_URIPATH, "/exhibitor");
+            conf.put(Config.STORM_EXHIBITOR_POLL, 0);
+            conf.put(Config.STORM_EXHIBITOR_RETRY_INTERVAL, 0);
+            conf.put(Config.STORM_EXHIBITOR_RETRY_INTERVAL_CEILING, 0);
+            conf.put(Config.STORM_EXHIBITOR_RETRY_TIMES, 0);
+        }
+        conf.put(Config.STORM_ZOOKEEPER_CONNECTION_TIMEOUT, 0);
+        conf.put(Config.STORM_ZOOKEEPER_SESSION_TIMEOUT, 0);
+        conf.put(Config.STORM_ZOOKEEPER_RETRY_INTERVAL, 0);
+        conf.put(Config.STORM_ZOOKEEPER_RETRY_INTERVAL_CEILING, 0);
+        conf.put(Config.STORM_ZOOKEEPER_RETRY_TIMES, 0);
+        String zkStr = new String("zk_connection_string");
+        ZookeeperAuthInfo auth = null;
+        if (withAuth) {
+            auth = new ZookeeperAuthInfo("scheme", "abc".getBytes());
+        }
+        Utils.testSetupBuilder(builder, zkStr, conf, auth);
+        return builder;
+    }
+
+    @Test
+    public void givenExhibitorServersBuilderUsesExhibitorProviderTest() {
+        CuratorFrameworkFactory.Builder builder = setupBuilder(true /*with exhibitor*/);
+        Assert.assertEquals(builder.getEnsembleProvider().getConnectionString(), "");
+        Assert.assertEquals(builder.getEnsembleProvider().getClass(), ExhibitorEnsembleProvider.class);
+    }
+
+    @Test
+    public void givenNoExhibitorServersBuilderUsesFixedProviderTest() {
+        CuratorFrameworkFactory.Builder builder = setupBuilder(false /*without exhibitor*/);
+        Assert.assertEquals(builder.getEnsembleProvider().getConnectionString(), "zk_connection_string");
+        Assert.assertEquals(builder.getEnsembleProvider().getClass(), FixedEnsembleProvider.class);
+    }
+
+    @Test
+    public void givenSchemeAndPayloadBuilderUsesAuthTest() {
+        CuratorFrameworkFactory.Builder builder = setupBuilder(false /*without exhibitor*/, true /*with auth*/);
+        List<AuthInfo> authInfos = builder.getAuthInfos();
+        AuthInfo authInfo = authInfos.get(0);
+        Assert.assertEquals(authInfo.getScheme(), "scheme"); 
+        Assert.assertArrayEquals(authInfo.getAuth(), "abc".getBytes());
+    }
+
+    @Test
+    public void parseJvmHeapMemByChildOptsTest() {
+        Assert.assertEquals(
+            "1024K results in 1 MB",
+            Utils.parseJvmHeapMemByChildOpts("Xmx1024K", 0.0).doubleValue(), 1.0, 0); 
+
+        Assert.assertEquals(
+            "100M results in 100 MB",
+            Utils.parseJvmHeapMemByChildOpts("Xmx100M", 0.0).doubleValue(), 100.0, 0); 
+
+        Assert.assertEquals(
+            "1G results in 1024 MB",
+            Utils.parseJvmHeapMemByChildOpts("Xmx1G", 0.0).doubleValue(), 1024.0, 0); 
+
+        Assert.assertEquals(
+            "Unmatched value results in default",
+            Utils.parseJvmHeapMemByChildOpts("Xmx1T", 123.0).doubleValue(), 123.0, 0); 
```

Comment:
I left this in since it's testing that what happens when parseJvmHeapMemByChildOpts can't parse it. I can make it something else if preferred.